File: js/algorithms/bubble.js
var utils = require('./utils/array'),
    swap = utils.swap;

function sort(array) {
    var last = array.length - 1,
        sorted,
        i,
        j;
    
    for (i = 0; i <= last; i++) {
        sorted = true;
        
        for (j = last; j > i; j--) {

            if (array[j] < array[j - 1]) {
                swap(array, j, j - 1);
                sorted = false;
            }

        }
        
        if (sorted) {
            break;
        }

    }
    
}


exports.name = 'bubble';
exports.sort = sort;

File: c++/algorithms/bubble/bubble.cpp
#include "bubble.h"

void BubbleSort::sort(int *array, int length) {

	int  i, current, next;
	bool sorted;

	for (i = length - 1; i >= 0; i--) {
		
		sorted = true;
		
		for (current = 0, next = 1; current < i; current = next, next++) {
			if (array[current] > array[next]) {
				swap(array, current, next);
				sorted = false;
			}
		}

		if (sorted) {
			break;
		}

	}

}
